ledger/ledgercore/persistedacctdata.go

```diff
@@ -116,3 +117,458 @@ var EncodedMaxAppLocalStates = basics.EncodedMaxAppLocalStates
 
 // EncodedMaxAppParams is a copy from basics package to resolve deps in msgp-generated file
 var EncodedMaxAppParams = basics.EncodedMaxAppParams
+
+// NumAssetHoldings returns number of assets in the account
+func (pad PersistedAccountData) NumAssetHoldings() int {
+	if pad.ExtendedAssetHolding.Count > 0 {
+		return int(pad.ExtendedAssetHolding.Count)
+	}
+	return len(pad.AccountData.Assets)
+}
+
+func (gd *AssetsHoldingGroupData) update(ai int, hodl basics.AssetHolding) {
+	gd.Amounts[ai] = hodl.Amount
+	gd.Frozens[ai] = hodl.Frozen
+}
+
+func (gd *AssetsHoldingGroupData) delete(ai int) {
```

Comment:
add a comment like "delete the ai-th element in the group holding array. The method expect the ai to be a valid index"